src/slices/chartSlice.ts

```diff
@@ -87,8 +87,26 @@ const chartSlice = createSlice({
                 cursorEnd: null | number;
             }>
         ) => {
-            state.cursorBegin = action.payload.cursorBegin;
-            state.cursorEnd = action.payload.cursorEnd;
+            const end =
+                action.payload.cursorEnd != null
+                    ? Math.max(
+                          0,
+                          Math.min(
+                              DataManager().getTimestamp(),
+                              action.payload.cursorEnd
+                          )
+                      )
+                    : null;
+            const begin =
+                action.payload.cursorBegin != null
+                    ? Math.min(
+                          Math.max(0, action.payload.cursorBegin),
+                          DataManager().getTimestamp()
+                      )
+                    : null;
+
+            state.cursorBegin = begin;
+            state.cursorEnd = end;
```

Comment:
Can we ask ourselves, "do we ever want one of `cursorBegin` or `cursorEnd` to be null while the other is not?

I think that in the `chartSlice` we only want either both to be `null` or both to be a `number`. And in that case, we could set both to `null` if either of them are `null`, and then I would prefer something like this, because it makes it clear that they either both ar `null` or both are `number`:

```typescript
        chartCursorAction: (
            state,
            {
                payload: { cursorBegin, cursorEnd },
            }: PayloadAction<{
                cursorBegin: null | number;
                cursorEnd: null | number;
            }>
        ) => {
            if (cursorBegin == null || cursorEnd == null) {
                state.cursorBegin = null;
                state.cursorEnd = null;
                return;
            }

            state.cursorBegin = Math.max(
                0,
                Math.min(DataManager().getTimestamp(), cursorEnd)
            );
            state.cursorEnd = Math.min(
                Math.max(0, cursorBegin),
                DataManager().getTimestamp()
            );
        },
```